docs/tutorial/index.md

```diff
@@ -105,135 +145,155 @@ You can also watch juju logs with the [`juju debug-log`](https://juju.is/docs/ju
 
-Check the IP addresses associated with each application unit with the `juju status` command. 
+```{terminal}
+:input: juju add-secret tutorial operator=mypassword
+:user: ubuntu
+:host: my-vm
 
-Since we will use the leader unit to connect to PostgreSQL, we are interested in the IP address for the unit marked with `*`, like in the output below:
+secret:d1ohj30ek0fco390bt9g
 
-```text
-Unit           	  Workload  Agent  Address   Ports  Message
-postgresql/0*     active	idle   10.1.110.80     	Primary
+:input: juju grant-secret tutorial postgresql
 ```
 
-The user we will connect to in this tutorial will be 'operator'. To retrieve its associated password, run the juju action `get-password`:
-
-```text
-juju run postgresql/leader get-password
+```{seealso}
+For more information about password management with Juju secrets, see [](/how-to/manage-passwords)
 ```
-The command above should output something like this:
-```text
-Running operation 1 with 1 task
-  - task 2 on unit-postgresql-0
 
-Waiting for task 2...
-password: 66hDfCMm3ofT0yrG
-```
-In order to retrieve the password of a user other than `operator`, use the `username` option:
+One more step is needed for the charm to update the passwords of its internal users based on our new Juju secret: we need to update the charm's `system-users` config option:
 
-```text
-juju run postgresql/leader get-password username=replication
+```{terminal}
+:input: juju config postgresql system-users=secret:d1ohj30ek0fco390bt9g
+:user: ubuntu
+:host: my-vm
 ```
 
-At this point, we have all the information required to access PostgreSQL. Run the command below to enter the leader unit's shell as root:
-
-```text
-juju ssh --container postgresql postgresql/leader bash
+```{tip}
+Remember to replace the secret URI above with yours! 
 ```
 
-which should bring you to a prompt like this: 
+Now we have all the information required to access PostgreSQL. Run the command below to enter the leader unit's shell:
 
-```text
- root@postgresql-0:/#
-```
-
-```{tip}
-If you’d like to leave the unit's shell and return to your local terminal, enter `Ctrl+D` or type `exit`.
+```{terminal}
+:input: juju ssh --container postgresql postgresql/leader bash
+:user: ubuntu
+:host: my-vm
 ```
 
 ### Create a database
 
 The easiest way to interact with PostgreSQL is via [PostgreSQL interactive terminal `psql`](https://www.postgresql.org/docs/14/app-psql.html), which is already installed on the host you're connected to.
 
-While still in the leader unit's shell, run the command below to list all databases currently available:
+We'll need the IP address associated with the specific application unit we want to interact with. You can find it with `juju status`. 
+
+Since we will use the leader unit to connect to PostgreSQL, we are interested in the address for the unit marked with `*`, like in the output below:
```

Comment:
We should most probably instruct users to look for the primary, not the juju leader. It shouldn't matter for the tutorial, but in principle the leader and thee primary could be different.